src/animated/Controller.ts

```diff
@@ -273,20 +279,28 @@ class Controller<DS extends object = any> {
         })
       })
     } else if (is.fun(to)) {
+      // Throw this to interrupt an infinite script
+      const stopToken = {}
+
       let i = 0
       let last: Promise<any>
       queue = queue.then(() =>
         to(
           // next(props)
           p => {
-            if (guid !== this.guid) return
+            if (guid !== this.guid) throw stopToken
```

Comment:
Bail out of async scripts when (1) new updates were applied, or (2) the `stop` method was called.